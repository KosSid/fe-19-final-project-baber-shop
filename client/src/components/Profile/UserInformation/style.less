@import '~antd/lib/style/themes/default.less';

#userInfo {
  & .ant-form-item {
    max-width: 700px;
  }

  & .ant-form-item-required {
    font-weight: bold;
    font-size: 14px;

    &::before {
      content: none;
    }
  }

  & .ant-input {
    border: 1px solid @black;


    &:hover {
      border: 1px solid @yellow-5;
    }

    &:focus {
      border: 1px solid @black;
      box-shadow: 0 0 8px @yellow-5;
    }
  }

  & .ant-form-item-control-input-content {
    text-align: end;
  }

  & button {
    min-width: 10vw;
    min-height: 2.7vw;
    color: @black;
    font-size: 14px;
    text-shadow: none;
    background: @yellow-5;
    border: 1px solid @yellow-5;

    &:hover {
      color: @black;
      background: @yellow-6;
      border: 1px solid @black;
    }
  }
}

@media (min-width: 768px) {
  #userInfo {
    & .ant-form-item-required {
      font-size: 16px;
    }

    & button {
      font-size: 16px;
    }
  }
}


@media (min-width: 1200px) {
  #userInfo {
    & .ant-form-item-required {
      font-size: 20px;
    }

    & button {
      min-width: 130px;
      min-height: 40px;
      font-size: 20px;
    }
  }
}
