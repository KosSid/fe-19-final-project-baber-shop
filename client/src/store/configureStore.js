import {createStore, compose, applyMiddleware} from "redux";
import rootReducer from "./rootReducer";
import thunk from "redux-thunk";
import jwt_decode from "jwt-decode";
import {authUser} from "./user/userAction";
import axios from "axios";
import { persistStore } from 'redux-persist'

const checkTokenExpirationMiddleware = store => next => action => {
    const {dispatch} = store;
    if (localStorage.getItem("token")) {
        const decoded = jwt_decode(localStorage.getItem("token"));
        if (decoded && decoded.exp && (decoded.exp < Date.now() / 1000)) {
            localStorage.removeItem("token");
            dispatch(authUser({isAdmin: false}));
        }
    }
    next(action);
};

const setAxiosHeaders = store => next => action => {

    const token = localStorage.getItem('token');

    if (token){
        axios.defaults.headers.common.Authorization = token;
    } else {
        delete axios.defaults.headers.common.Authorization;
    }
    next(action);
}


const devTools = window.__REDUX_DEVTOOLS_EXTENSION__
    ? window.__REDUX_DEVTOOLS_EXTENSION__()
    : (f) => f;

export const store = createStore(
    rootReducer,
    compose(applyMiddleware(thunk, checkTokenExpirationMiddleware, setAxiosHeaders), devTools)
);
export const persistor = persistStore(store)

export default {store, persistor};
