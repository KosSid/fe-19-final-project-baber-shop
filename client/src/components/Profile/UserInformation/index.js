import React, {useEffect} from "react";
import {useDispatch} from "react-redux";
import {Button, Col, Form, Input, message} from "antd";
import Ajax from "../../../services/Ajax";
import {changeFirstName} from "../../../store/user/userAction";
import {collectionItemsForm, onlyLetters, onlyNumbers} from "../../Forms/RegistrationForm/collectionItems";
import './style.less'

const {get, put} = Ajax;


const UserInformation = () => {
    const dispatch = useDispatch();

    const layout = {
        labelCol: {
            span: 24,
        },
        wrapperCol: {
            span: 24
        },
    };

    const formTailLayout = {
        labelCol: {
            span: 24,
        },
        wrapperCol: {
            span: 24,
        },
    };
    const [form] = Form.useForm();
    const success = () => {
        message.success('Your user data has been successfully updated');
    };
    const error = () => {
        message.error('Enter correct information');
    };
    const onFinish = async () => {
        try {
            const values = await form.validateFields();
            await put('/customers', '', values);
            console.log(values);
            dispatch(changeFirstName(values.firstName));
            success();

        } catch (errorInfo) {
            error();
        }
    }

    useEffect(() => {
        let cleanupFunction = false;
        get('/customers/customer')
            .then(customer => {
                if (!cleanupFunction)
                    form.setFieldsValue({
                            firstName: customer.firstName,
                            lastName: customer.lastName,
                            login: customer.login,
                            email: customer.email,
                            phone: customer.phone,
                        }
                    )

            })
        return () => cleanupFunction = true
    }, [form])

    return (
        <Col xs={{span: 20, offset: 2}} sm={{span: 12, offset: 1}} md={{span: 12, offset: 2}}
             xl={{span: 10, offset: 2}}>
            <Form
                form={form}
                {...layout}
                name="userInfo"
                initialValues={{
                    remember: true,
                }}
            >
                {collectionItemsForm.map(formItem =>
                    formItem.name !== "password"
                        ? <Form.Item
                            label={formItem.label}
                            name={formItem.name}
                            rules={formItem.rules}
                            key={formItem.name}
                        >
                            {formItem.name === "phone"
                                ? <Input maxLength={13} onKeyPress={onlyNumbers()}/>
                                : formItem.name === "firstName" || formItem.name === "lastName"
                                    ? <Input placeholder={formItem.label} onKeyPress={onlyLetters()} maxLength={25}/>
                                    : <Input placeholder={formItem.label} maxLength={25}/>
                            }
                        </Form.Item>
                        : ''
                )}
                <Form.Item {...formTailLayout}>
                    <Button type="primary" onClick={onFinish}>
                        Submit
                    </Button>
                </Form.Item>
            </Form>
        </Col>
    )
}
export default UserInformation;