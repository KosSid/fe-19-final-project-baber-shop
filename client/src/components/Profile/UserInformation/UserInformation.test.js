import React from 'react'
import {render, unmountComponentAtNode} from 'react-dom'
import {act} from '@testing-library/react'
import {MemoryRouter} from "react-router";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store'
import UserInformation from "./index";


let container = null;
const initialState = {output:10}
const mockStore = configureStore()
let store

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
})

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
})

test('UserInformation is rendered true', () => {
    store = mockStore(initialState)
    act(() => {
        render(
            <Provider store={store}>
                <MemoryRouter>
                    <UserInformation/>
                </MemoryRouter>
            </Provider>, container)
    });

    expect(container.innerHTML).toMatchSnapshot();
});


test("UserInformation confirm button called Submit", () => {
    store = mockStore(initialState)
    act(() => {
        render(
            <Provider store={store}>
                <MemoryRouter>
                    <UserInformation/>
                </MemoryRouter>
            </Provider>, container)
    });

    const text = document.querySelector("button > span");
    console.log(text.innerHTML)
    expect(text.innerHTML).toBe("Submit");

});