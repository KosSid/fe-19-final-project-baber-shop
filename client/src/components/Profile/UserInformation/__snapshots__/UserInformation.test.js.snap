// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInformation is rendered true 1`] = `"<div class=\\"ant-col ant-col-xs-20 ant-col-xs-offset-2 ant-col-sm-12 ant-col-sm-offset-1 ant-col-md-12 ant-col-md-offset-2 ant-col-xl-10 ant-col-xl-offset-2\\"><form id=\\"userInfo\\" class=\\"ant-form ant-form-horizontal\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-24 ant-form-item-label\\"><label for=\\"userInfo_firstName\\" class=\\"ant-form-item-required\\" title=\\"First Name\\">First Name</label></div><div class=\\"ant-col ant-col-24 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><input placeholder=\\"First Name\\" maxlength=\\"25\\" type=\\"text\\" id=\\"userInfo_firstName\\" class=\\"ant-input\\" value=\\"\\"></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-24 ant-form-item-label\\"><label for=\\"userInfo_lastName\\" class=\\"ant-form-item-required\\" title=\\"Last Name\\">Last Name</label></div><div class=\\"ant-col ant-col-24 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><input placeholder=\\"Last Name\\" maxlength=\\"25\\" type=\\"text\\" id=\\"userInfo_lastName\\" class=\\"ant-input\\" value=\\"\\"></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-24 ant-form-item-label\\"><label for=\\"userInfo_login\\" class=\\"ant-form-item-required\\" title=\\"Login\\">Login</label></div><div class=\\"ant-col ant-col-24 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><input placeholder=\\"Login\\" maxlength=\\"25\\" type=\\"text\\" id=\\"userInfo_login\\" class=\\"ant-input\\" value=\\"\\"></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-24 ant-form-item-label\\"><label for=\\"userInfo_email\\" class=\\"ant-form-item-required\\" title=\\"E-mail\\">E-mail</label></div><div class=\\"ant-col ant-col-24 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><input placeholder=\\"E-mail\\" maxlength=\\"25\\" type=\\"text\\" id=\\"userInfo_email\\" class=\\"ant-input\\" value=\\"\\"></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-24 ant-form-item-label\\"><label for=\\"userInfo_phone\\" class=\\"ant-form-item-required\\" title=\\"Phone Number\\">Phone Number</label></div><div class=\\"ant-col ant-col-24 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><input maxlength=\\"13\\" type=\\"text\\" id=\\"userInfo_phone\\" class=\\"ant-input\\" value=\\"+380\\"></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-24 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><button type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Submit</span></button></div></div></div></div></form></div>"`;
